### DISCUSSION TITLE
Getting a Rich style from a CSS colour variable?

### DISCUSSION THREAD

**Discussion by mon** [Textualize/textual#4166] [Q&A] [answered]
I have a component that I want to have a dimmed prefix before normal text.
Because I'm using `render` instead of `compose`, I can't just slap two `Static`s together and call it a day.

My current approach seems awfully complex:
- `DEFAULT_CSS` which specifies two new classes, one with `color: $text-disabled` and one with `color: $text`
- `COMPONENT_CLASSES` referencing those two new classes
... and finally, the text itself:
```python
self.value = Text(
    "0x", style=self.get_component_rich_style("mywidget--prefix")
).append(
    f"{hex_value:X}",
    style=self.get_component_rich_style("mywidget--normal"),
)
```

Is there an easier way around this? In an idea world I'd like to just write:
```python
self.value = Text(
    "0x", style=rich_style_from_css("color: $text-disabled")
).append(
    f"{hex_value:X}",
    style=rich_style_from_css("color: $text")
)
```

**Comment by davep** [comment, marked as answer]
Normally when I want to do this I just use the `[dim]` Rich markup. For example [this code](https://github.com/davep/tinboard/blob/0a11c3fb7f90c1ebc091808a7875b6de70e57bb4/tinboard/widgets/bookmarks.py#L78-L97) is what's involved in making this:

![Screenshot 2024-02-19 at 08 52 42](https://github.com/Textualize/textual/assets/28237/09ec824a-bca7-47df-bb04-cd9a0478baea)

**Reply by mon** [reply]
Oh, that's super convenient! Does it respect light/dark theme? I guess a medium grey is probably about the same regardless of the background 😅

**Reply by davep** [reply]
Good question. Not sure about "respect", but it seems to look fine in my apps:

![Screenshot 2024-02-19 at 09 01 02](https://github.com/Textualize/textual/assets/28237/4ec6b4b9-6c11-4221-abea-6444f730b020)

**Reply by mon** [reply]
Totally usable in my mind, and the rich markup is super compact. Works for me!
